File: examples/python/run_from_yaml.py
"""
run_from_yaml.py — Minimal example: load a robot from YAML and solve IK.

Usage (from the repository root):
    python3 examples/python/run_from_yaml.py
"""

from _common import print_solutions
import ikdh

# ── Load robot from YAML ──────────────────────────────────────────────────────
robot  = ikdh.load_robot("robots/gofa5.yaml")
solver = ikdh.Solver(robot.dh, robot.limits)

# ── Solve IK for two end-effector poses ───────────────────────────────────────
# pose_from_xyzrpw: x y z in mm, rx ry rz in degrees (RoboDK convention Rz*Ry*Rx)
poses = [
    ikdh.pose_from_xyzrpw(200.0, 0.0, 600.0,   0.0, 90.0, 0.0),
    ikdh.pose_from_xyzrpw(400.0, 0.0, 300.0, 180.0,  0.0, 0.0),
]

for ee in poses:
    print_solutions(solver, robot.dh, ee)

File: examples/cpp/run_from_yaml.cpp
#include <ikdh.h>
#include <robots.h>
#include "example_common.h"

#ifndef ROBOTS_DIR
#  define ROBOTS_DIR "robots"
#endif

int main()
{
    // ── Load robot from YAML ──────────────────────────────────────────────────
    auto robot = Robots::loadRobot(ROBOTS_DIR "/gofa5.yaml");
    IKDH::Solver solver(robot.dh, robot.limits);

    // ── Solve IK for two end-effector poses ───────────────────────────────────
    // poseFromXYZRPW: x y z in mm, rx ry rz in degrees (RoboDK convention Rz*Ry*Rx)
    IKDH::Transform poses[] = {
        IKDH::poseFromXYZRPW(200.0, 0.0, 600.0,   0.0, 90.0, 0.0),
        IKDH::poseFromXYZRPW(400.0, 0.0, 300.0, 180.0,  0.0, 0.0),
    };

    for (const auto& ee : poses) printSolutions(solver, robot.dh, ee);
}
